### ISSUE TITLE
BERT model for text classification for custom data training using my own trained BERT model not working

### ISSUE THREAD

**Issue by mgrove6** [ThilinaRajapakse/simpletransformers#800]
Hi,
I have trained a BERT text classification model for 10 classes. Now I want to train the same model again for a 1 or 2 new label classes. But when I load this trained model and use it to train on new classes I am getting this error-
 ' File "/Users/mani/Documents/BrickSimple/projects/Whitemarsh-Pro-MVP/visual-studio-stuff/env-text-classification-py3.7-torchversion/lib/python3.7/site-packages/simpletransformers/classification/classification_model.py", line 138, in __init__
    assert num_labels == len(self.args.labels_list)
AssertionError'
Could you please help me with this or tell me how can I use my own trained BERT model for new classes?
Thank you

**Comment by manueltonneau** [comment]
Hi @mgrove6, it seems the error comes from the fact that you first fine-tune BERT with X labels and then try to further fine-tune it with X+1 or X+2 labels. After the first fine-tuning, your model becomes a BertForSequenceClassification object with the number of labels defined as X. I'm not sure how you can easily change this number to further fine-tune the model.

**Comment by mgrove6** [comment]
Hi @manueltonneau thank you for this information. Do you have any idea how I can use my custom model's weights which was trained on X labels, when I train a new bert model for X+1 labels? Thank you.

**Comment by ThilinaRajapakse** [comment]
I'm with Manuel, I don't think you can change the number of labels like that. I can offer a couple of workarounds though. The simplest approach is to probably set your `num_labels` to X + 1 or X + 2 in the first stage even though you only have X labels. Another approach would be to manually load the weights for everything except the classification layer. You won't be able to load the model with Simple Transformers, but you can use Pytorch methods to load the weights after initializing your `ClassificationModel`. 

E.g. :

```python
model = ClassificationModel("bert", "bert-base-cased", args=model_args)  # Only using bert-base-cased to initialize the model

model.model.load_state_dict(torch.load(os.path.join(self.args.model_name, "pytorch_model.bin")))  # This won't work straightaway, see below.
```

You'll need to handle the mismatched keys (see [here](https://discuss.pytorch.org/t/how-to-load-part-of-pre-trained-model/1113/7)).
